src/qibo/result.py

```diff
@@ -55,31 +59,33 @@ def symbolic(self, decimals=5, cutoff=1e-10, max_terms=20):
             )
         return " + ".join(terms)
 
-    def state(self, numpy=False):
+    def state(self, numpy: bool = False):
         """State's tensor representation as a backend tensor.
 
         Args:
-            numpy (bool): If ``True`` the returned tensor will be a numpy array,
+            numpy (bool, optional): If ``True`` the returned tensor will be a numpy array,
                 otherwise it will follow the backend tensor type.
-                Default is ``False``.
+                Defaults to ``False``.
 
         Returns:
             The state in the computational basis.
         """
         if numpy:
-            return np.asarray(self._state)
-        else:
-            return self._state
+            return np.array(self._state)
```

Comment:
I haven't tested it, but this looks like it will fail if the original backend is one of the GPU backends.

Reply:
The tests with gpus were passing already I believe, but if we can get rid of an `if/else` fine by me.